python/nammu/view/AtfAreaView.py

```diff
@@ -402,8 +403,36 @@ def undoableEditHappened(self, event):
             # to false. Note undo() only undoes compound edits when they
             # are not in progress.
             self.current_compound.end()
-            self.current_compound = CompoundEdit()
+            self.current_compound = AtfCompoundEdit()
             self.undo_manager.addEdit(self.current_compound)
 
         # Always add current edit to current compound
         self.current_compound.addEdit(edit)
+
+
+class AtfCompoundEdit(CompoundEdit):
+    # add a getter for a protected field
+    def getEdits(self):
+        field = CompoundEdit.getDeclaredField("edits")
+        field.setAccessible(True)
+        return field.get(self)
+
+    def firstEdit(self):
+        return self.getEdits()[0]
+
+
+class AtfUndoManager(UndoManager):
+    def __init__(self, panel):
+        self.panel = panel
```

Comment:
Is this used anywhere?

Reply:
I just realised that I used this for debugging, with this member we can access the logger, but there is probably no need for it in production.  Should I remove it anyway?

Reply:
If you think it will be useful, I would leave it in but with a comment clarifying why (e.g. "this lets us access the logger if required for debugging") -- or, better yet, a docstring on the class explaining the differences from the base `UndoManager`, including how to log.

Or maybe there is another way of logging things about the undo manager from outside (if we need to)?

Reply:
> Or maybe there is another way of logging things about the undo manager from outside (if we need to)?

By creating a new logger.  It's not science rocket, it's just easier to just use an existing (and already configured) one.  I'm going to add a comment to explain this.